internal/role/p2p/k8s.go

```diff
@@ -37,21 +37,187 @@ type K8sNode interface {
 	Distro() string
 }
 
+// configState holds the parsed configuration state.
+type configState struct {
```

Comment:
```suggestion
type configExplicitState struct {
```

because otherwise it feels strange that we need both k0sEnabled and k0sDisabled (because one would be the opposite of the other). A comment explaining this might also help.

Reply:
@jimmykarily good point